modules/qna/src/views/list/ElementsList.jsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+import { ListGroupItem, Glyphicon } from 'react-bootstrap'
+import { InputElement } from './InputElement'
+import style from './style.scss'
+
+export class ElementsList extends React.Component {
+  elementInputRef = React.createRef()
+  state = {
+    inputValue: '',
+    error: undefined,
+    elements: [],
+    editElementIndex: undefined
+  }
+
+  createElement = element => {
+    this.setState({ elements: [...this.state.elements, element] }, this.onElementsChange)
+  }
+
+  updateElement = (element, index) => {
+    const elements = this.state.elements
+    if (elements[index]) {
+      elements[index] = element
+      this.setState({ elements, editElementIndex: undefined }, this.onElementsChange)
+    }
+  }
+
+  deleteElement = index => {
+    const elements = this.state.elements
+
+    if (elements[index]) {
+      elements.splice(index, 1)
+      this.setState({ elements }, this.onElementsChange)
+    }
+  }
+
+  toggleEditMode = index => {
+    this.setState({ editElementIndex: index })
+  }
+
+  onElementsChange = () => {
```

Comment:
I'll quote @allardy here, we use handle as convention ==> `handleElementsChange`.. really not an issue for me though

Reply:
In this case its really "event" like. I trigger `onElementsChange` which is a callback function that returns the new elements. It doesn't "handle" anything.